src/rpc/server.h

```diff
@@ -42,11 +42,25 @@ class JSONRPCRequest
 public:
     UniValue id;
     std::string strMethod;
+    /**
+     * Parameters from JSON-RPC request.
+     * This will be either an object or an array when the JSONRPCRequest object is
+     * originally created and parsed. But it will be transformed into an
+     * array before being passed to the RPC method implementation (using the
+     * list of named arguments provided by the implementation).
+     */
     UniValue params;
     bool fHelp;
     std::string URI;
     std::string authUser;
 
+    /**
+     * Optional wallet name, set for backwards compatibility if the RPC method
+     * was called with a named "wallet" parameter and the RPC method
+     * implementation doesn't handle it itself.
+     */
+    std::string wallet;
```

Comment:
> Wallets should be passed as a CWallet reference, not as a string...

Advantage of doing that is that it could simplify test code wanting to emulate RPC calls by avoiding the need for it to have to manipulate `vpwallets.` Disadvantage is it would require sticking an `#ifdef ENABLE_WALLET` in the PR and moving wallet-selection logic from the wallet layer (`GetWalletForJSONRPCRequest`) to the rpc layer (`transformNamedArguments`).